components/checkout/CheckoutConfirmation.tsx

```diff
@@ -62,54 +65,82 @@ const CheckoutConfirmation = ({
       return;
     }
 
-    const bookingId = generateBookingId();
-
-    const currentUrl =
-      typeof window !== "undefined" ? window.location.origin : "";
-    const returnUrl = `${currentUrl}/payment?bookingId=${bookingId}`;
-    const { error } = await stripe.confirmPayment({
-      elements,
-      clientSecret,
-      confirmParams: {
-        // Custom booking URL
-        return_url: returnUrl,
+    const FINAL_BOOKING_DETAILS: FINAL_BOOKING_INFO = {
+      email: session?.user.email!,
+      bookingId: generateBookingId(),
+      checkin_date: bookingDetails.checkin_date,
+      checkout_date: bookingDetails.checkout_date,
+      adults_number: bookingDetails.adults_number,
+      numDays: bookingDetails.numDays,
+      hotel_id: hotelRoomOffer.hotel_id,
+      room_id: hotelRoomOffer.hotel_room_id,
+      payment_info: {
+        firstName: paymentInfo.firstName,
+        lastName: paymentInfo.lastName,
+        billingAddress: paymentInfo.billingAddress,
+        city: paymentInfo.city,
+        state: paymentInfo.state,
+        zipCode: paymentInfo.zipCode,
+        email: paymentInfo.email,
       },
-    });
+      transaction_info: {
+        dateCreated: new Date().toISOString(),
+        stripePaymentId: "",
+      },
+    };
+
+    // Create reservation in DB
+    const PrismaReservationDB: PartialReservation = {
```

Comment:
It seems like FINAL_BOOKING_DETAILS has nearly the same schema as the object you're unpacking it into. Instead of verbosely mapping it, you can just create a deepcopy.

Reply:
I notice you do this a lot. Don't worry about fixing it in this PR.

Reply:
Yeah, we can address this in a future PR. I wanted to link the existing data with reservation.